docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,428 @@
+
+### Supported OAuth flow
+
+`wire-server` currently only supports the [authorization code flow](https://www.rfc-editor.org/rfc/rfc6749#section-4.1) which is optimized for confidential clients such as Outlook Calendar Extension.
+
+```{eval-rst}
+.. kroki::
+   :type: mermaid
+   :caption: OAuth 2.0 authorization code flow
+
+   sequenceDiagram
+       autonumber
+       actor U as User
+       participant C as Outlook Calendar Extension
+       participant A as Authorization Server (wire-server)
+       participant R as Resource Server (wire-server)
+
+       U->>C: Click login
+       C->>A: Authorization code request /authorize
+       A->>U: Redirect to login/authorization prompt
+       U->>A: Authenticate and consent
+       A->>C: Authorization code
+       C->>A: Authentication code + client credentials
+       A->>A: Validate authentication code + client credentials
+       A->>C: Access token
+       C->>R: Request a resource with access token (e.g. POST /conversations)
+       R->>R: Validate access token with public key from auth server
+       R->>C: Response
+```
+
+## OAuth client developer reference
+
+### Registering an OAuth client
+
+A new OAuth client can be register _only_ via the internal API of `brig` by providing an application name and a redirect URL:
+
+```curl
+  curl -s -X POST server-internal.example.com/i/oauth/clients \
+    -H "Content-Type: application/json" \
+    -d '{
+      "application_name":"Outlook Calendar Extension",
+      "redirect_url":"https://client.example.com"
+    }'
+```
+
+Parameters:
+
+| Parameter          | Description                                                                                          |
+| ------------------ | ---------------------------------------------------------------------------------------------------- |
+| `redirect_url`     | The URL to which Wire app will redirect the browser after authorization has been granted by the user |
+| `application_name` | The name of the application that will be shown on the consent page                                   |
+
+Client credentials will be generated and returned by wire-server:
+
+```json
+{
+  "clientId": "b9e65569-aa61-462d-915d-94c8d6ef17a7",
+  "clientSecret": "3f6fbd62835859b2bac411b2a2a2a54699ec56504ee32099748de3a762d41a2d"
+}
+```
+
+These credentials have to be stored in a safe place and cannot be recovered if they are lost.
+
+### Get authorization code
+
+When the user wants to use the 3rd party app for the first time, they need to authorize it to access Wire resources on their behalf.
+
+To do so the user is redirected to a login screen for authentication. Once authenticated, the user can either grant or deny the client's access request and the corresponding scope (a list of permissions to give to the 3rd party app).
```

Comment:
is that the wire login screen or the google one?  if it's the wire one, please say so.  also: why do we redirect to the login screen?  odds are the user is already logged in and could be redirected to the oauth page directly.  also, how does the user get to the grant page after login?